### PR TITLE
Add .. seealso:: in Study.get_trials and Study.trials to refer to each other #3578

### PR DESCRIPTION
**Pull request by jmsykes83**
## Motivation
Add .. seealso:: in Study.get_trials and Study.trials to refer to each other #3578
Issue: #3578

## Description of the changes
Updated the dosctrings for Study.trials and Study.get_trials.  To reference each other.

### CODE REVIEW

**Comment by c-bata** [comment]
@jmsykes83 Thank you for your pull request. Could you fix the documentation build error?

**Comment by jmsykes83** [comment]
Yes,  I am working on it.

File: optuna/study/study.py [hunk 1]
```diff
@@ -210,6 +210,9 @@ def trials(self) -> List[FrozenTrial]:
 
         Returns:
             A list of :class:`~optuna.trial.FrozenTrial` objects.
+            -- seealso::
+                See :func:`~optuna.study.Study.get_trails` for related method.
+
```
Comment (nzw0301) [inline optuna/study/study.py:215]: Typo and fix syntax.
```suggestion

            .. seealso::
                See :func:`~optuna.study.Study.get_trials` for related method.

```

File: optuna/study/study.py [hunk 2]
```diff
@@ -222,6 +225,8 @@ def get_trials(
         """Return all trials in the study.
 
         The returned trials are ordered by trial number.
+        -- seealso::
+            See :func:`~optuna.study.Study.trails` for related method.
```
Comment (nzw0301) [inline optuna/study/study.py:229]: - Fix syntax: replace `--` with `::`
- Fix typo: trail -> trial 
- Replace `func` with `attr` 

```suggestion

        .. seealso::
            See :attr:`~optuna.study.Study.trials` for related method.
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**